internal/support/notifications/v2/application/sendingservice.go

```diff
@@ -0,0 +1,160 @@
+	transRecord.Status = models.Sent
+	switch channel.GetBaseAddress().Type {
+	case v2.REST:
+		restAddress, ok := channel.(models.RESTAddress)
+		if !ok {
+			return transRecord, errors.NewCommonEdgeX(errors.KindContractInvalid, "fail to cast Address to RESTAddress", nil)
+		}
+		transRecord.Response, err = utils.SendRequestWithRESTAddress(lc, n.Content, n.ContentType, restAddress)
+		if err != nil {
+			transRecord.Status = models.Failed
+			transRecord.Response = err.Error()
+		}
+		transRecord.Sent = common.MakeTimestamp()
+	case v2.EMAIL:
+		emailAddress, ok := channel.(models.EmailAddress)
+		if !ok {
+			return transRecord, errors.NewCommonEdgeX(errors.KindContractInvalid, "fail to cast Address to EmailAddress", nil)
+		}
+		transRecord.Response, err = utils.SendEmailWithAddress(lc, n.Content, n.ContentType, emailAddress)
+		if err != nil {
+			transRecord.Status = models.Failed
+			transRecord.Response = err.Error()
+		}
+		transRecord.Sent = common.MakeTimestamp()
+	default:
+		transRecord.Response = fmt.Sprintf("unsupported address type: %s", channel.GetBaseAddress().Type)
+	}
+	return transRecord, nil
+}
+
+// normalSend handles the notification transmission
+func normalSend(dic *di.Container, n models.Notification, trans models.Transmission) (models.Transmission, errors.EdgeX) {
+	lc := container.LoggingClientFrom(dic.Get)
+
+	record, err := sendNotificationViaChannel(lc, n, trans.Channel)
+	if err != nil {
+		return trans, errors.NewCommonEdgeXWrapper(err)
+	}
+	trans.Records = append(trans.Records, record)
+	trans.Status = record.Status
+	lc.Debugf("send the notification to %s with address %v, transmission status %s", trans.SubscriptionName, trans.Channel.GetBaseAddress(), trans.Status)
+	return trans, nil
+}
+
+func resendLimitAndInterval(config *config.ConfigurationStruct, sub models.Subscription) (int, time.Duration, errors.EdgeX) {
+	resendLimit := config.ResendLimit
+	if sub.ResendLimit > 0 {
+		resendLimit = sub.ResendLimit
+	}
+	resendInterval := config.ResendInterval
+	if sub.ResendInterval != "" {
+		resendInterval = sub.ResendInterval
+	}
+	resendIntervalDuration, err := time.ParseDuration(resendInterval)
+	if err != nil {
+		return 0, time.Second, errors.NewCommonEdgeX(errors.KindContractInvalid, "fail to parse resendInterval", err)
+	}
+	return resendLimit, resendIntervalDuration, nil
+}
+
+// criticalSend handles the Critical notification Transmission
+func criticalSend(dic *di.Container, n models.Notification, sub models.Subscription, trans models.Transmission) (models.Transmission, errors.EdgeX) {
+	dbClient := v2NotificationsContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+	config := notificationContainer.ConfigurationFrom(dic.Get)
+
+	resendLimit, resendInterval, err := resendLimitAndInterval(config, sub)
+	if err != nil {
+		return trans, errors.NewCommonEdgeXWrapper(err)
+	}
+	for i := 1; i <= resendLimit; i++ {
+		time.Sleep(resendInterval)
```

Comment:
Why sleeping before sending first attempt? Move to after attempts fail

Reply:
I moved the warning log, the sequence is:

- Send notification 
- Send again if failed and notification severity is critical

And the criticalSend() func will do
- Sleep and retry
- Send critical notification
- Sleep and retry
- Send critical notification

Reply:
Please at `why` comment since it is unusual to have the `Sleep` be the first thing in the loop.

Reply:
Added comment.